File: cuenta.py
class Cuenta:

    def __init__(self, saldo):
        self.__saldo = saldo

    def consultar_saldo(self):
        return self.__saldo

    def extraer_dinero(self, monto):
        if monto <= 0:
            return False

        if monto > self.__saldo:
            return False

        self.__saldo -= monto
        return True

File: src/models/cuenta.ts
export class Cuenta {
  // Encapsulamiento privado (__saldo en Python)
  private __saldo: number;

  constructor(saldoInicial: number = 5000000) {
    this.__saldo = Number(saldoInicial);
  }

  consultar_saldo(): number {
    return this.__saldo;
  }

  depositar(monto: number): void {
    const cantidad = Number(monto);
    if (cantidad > 0) {
      this.__saldo += cantidad;
    }
  }

  extraer(monto: number): boolean {
    const cantidad = Number(monto);
    if (cantidad > 0 && cantidad <= this.__saldo) {
      this.__saldo -= cantidad;
      return true;
    }
    return false;
  }

  reset(saldo: number = 5000000): void {
    this.__saldo = Number(saldo);
  }
}
